include/chopper/pack/hibf_statistics.hpp

```diff
@@ -0,0 +1,259 @@
+                      << s.num_ibfs << '\t'
+                      << to_formatted_BF_size(level_size) << '\t'
+                      << to_formatted_BF_size(level_size_no_corr) << '\t'
+                      << total_num_tbs << '\t'
+                      << total_num_tbs / s.num_ibfs << '\t'
+                      << split_tb_percentage << '\t';
+
+            // if there are no split bins on this level, the following statistics don't make sense
+            if (max_split_bin_span != 0)
+            {
+                size_t const total_num_split_ubs = std::reduce(s.num_split_ubs.begin(), s.num_split_ubs.end());
+                double const avg_split_bin = static_cast<double>(total_num_split_tbs)
+                                           / static_cast<double>(total_num_split_ubs);
+                size_t const total_split_tb_kmers = std::reduce(s.split_tb_kmers.begin(), s.split_tb_kmers.end());
+                double const avg_factor = static_cast<double>(std::reduce(s.split_tb_corr_kmers.begin(),
+                                                                          s.split_tb_corr_kmers.end()))
+                                        / static_cast<double>(total_split_tb_kmers);
+
+                std::cout << max_split_bin_span << '\t'
+                          << avg_split_bin << '\t'
+                          << (*fp_correction)[max_split_bin_span] << '\t'
+                          << avg_factor << '\n';
+            }
+            else
+            {
+                std::cout << "-\t-\t-\t-\n";
+            }
+        }
+
+        std::cout << "\tTotal HIBF size: " << to_formatted_BF_size(total_size)
+                  << "\n\tTotal HIBF size no correction: " << to_formatted_BF_size(total_size_no_corr) << "\n\n";
+    }
+
+    //!\brief The top level IBF of this HIBF, often starting point for recursions.
+    ibf top_level_ibf;
+
+private:
+    //!\brief Copy of the cser configuration for this HIBF.
+    pack_config const config{};
+
+    //!\brief The false positive correction factors to use for the statistics.
+    std::vector<double> const * const fp_correction{nullptr};
+
+    //!\brief Statistics for all IBFs on a certain level of the HIBF.
+    struct level_summary
+    {
+        size_t num_ibfs{};
+
+        std::vector<size_t> num_tbs{};
+        std::vector<size_t> num_ubs{};
+
+        std::vector<size_t> num_split_tbs{};
+        std::vector<size_t> num_merged_tbs{};
+
+        std::vector<size_t> num_split_ubs{};
+        std::vector<size_t> num_merged_ubs{};
+
+        std::vector<size_t> max_split_tb_span{};
+        std::vector<size_t> split_tb_corr_kmers{};
+        std::vector<size_t> split_tb_kmers{};
+
+        std::vector<size_t> max_ubs_in_merged{};
+
+        std::vector<size_t> ibf_mem_size{};
+        std::vector<size_t> ibf_mem_size_no_corr{};
+    };
+
+    //!\brief The gathered summary of statistics for each level of this HIBF.
+    std::map<size_t, level_summary> summary;
```

Comment:
Couldn't this just be a `std::vector<level_summary>`?
I guess we would need to know how big this will get?

Reply:
```suggestion
    std::map<size_t, level_summary> summaries;
```